src/transaction/transaction_master_group_complete_manager.cpp

```diff
@@ -159,14 +181,27 @@ namespace cubtx
 	return;
       }
 
+    while (!is_latest_closed_group_prepared_for_complete ())
+      {
+	/* It happens rare. */
+	thread_sleep (10);
+      }
+
+    if (!starts_latest_closed_group_complete ())
```

Comment:
I think is easier to understand if starts_latest_closed_group_complete is renamed as set_prev_group_complete_started
and returns boolean which says if we (current caller) need to complete the group (returning false means already performed by other thread)